lib/data_layer.ex

```diff
@@ -2120,35 +2120,63 @@ defmodule AshPostgres.DataLayer do
                 {Map.take(r, keys), r}
               end)
 
-            results =
-              changesets
-              |> Enum.map(fn changeset ->
-                identity =
-                  changeset.attributes
-                  |> Map.take(keys)
+            cant_map_results_to_changesets =
+              any_generated_keys_missing?(keys, resource, changesets)
```

Comment:
but I don't see how this is possible? Don't we require the keys to be given to do an upsert on them? And this is only inside the upsert branch.

Reply:
Oh, then we should switch back to index based when it's not an upsert. If it's just a create, either they are all created or not, there is no variability. Can you make that change?